Zend/zend_vm_execute.h

```diff
@@ -1772,6 +1772,144 @@ static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_FETCH_CLASS_NAME_SPEC_HANDLER(
 	ZEND_VM_NEXT_OPCODE();
 }
 
+static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_CALL_TRAMPOLINE_SPEC_HANDLER(ZEND_OPCODE_HANDLER_ARGS)
+{
+	zend_array *args;
+	zend_function *fbc = EX(func);
+	zend_object *object = Z_OBJ(EX(This));
+	zval *ret = EX(return_value);
+	zend_call_kind call_kind = EX_CALL_KIND();
+	zend_class_entry *scope = EX(called_scope);
+	uint32_t num_args = EX_NUM_ARGS();
+	zend_execute_data *call;
+	USE_OPLINE
+
+	args = emalloc(sizeof(zend_array));
+	zend_hash_init(args, num_args, NULL, ZVAL_PTR_DTOR, 0);
+	if (num_args) {
+		zval *p = ZEND_CALL_ARG(execute_data, 1);
+		zval *end = p + num_args;
+
+		zend_hash_real_init(args, 1);
+		ZEND_HASH_FILL_PACKED(args) {
+			do {
+				ZEND_HASH_FILL_ADD(p);
+				p++;
+			} while (p != end);
+		} ZEND_HASH_FILL_END();
+	}
+
+	SAVE_OPLINE();
+	call = execute_data;
+	execute_data = EG(current_execute_data) = EX(prev_execute_data);
+	zend_vm_stack_free_call_frame(call);
+	call = zend_vm_stack_push_call_frame(call_kind, fbc->common.prototype, 2, scope, object, execute_data);
+
+	ZVAL_STR(ZEND_CALL_ARG(call, 1), fbc->common.function_name);
+	ZVAL_ARR(ZEND_CALL_ARG(call, 2), args);
+	zend_free_trampoline(fbc);
+	fbc = call->func;
+
+	if (EXPECTED(fbc->type == ZEND_USER_FUNCTION)) {
+
+		ZEND_ASSERT(!(fbc->common.fn_flags & ZEND_ACC_GENERATOR));
+
+		EG(scope) = fbc->common.scope;
```

Comment:
EG(scope) is already set properly by previously call to this function (trampoline op_array). we don't need set it again here...